types/Types.d.ts

```diff
@@ -0,0 +1,369 @@
+interface CommonPlaidLinkOptions {
+    onSuccess: Function;
+    onExit?: Function;
+    logLevel?: LinkLogLevel;
+    extraParams?: Record<string, any>;
+}
+
+export type LinkTokenConfiguration = (CommonPlaidLinkOptions & {
+    token: string;
+});
+
+export type LinkPublicKeyConfiguration = (CommonPlaidLinkOptions & {
+    publicKey: string;
+    token?: string;
+    clientName: string;
+    environment: PlaidEnvironment;
+    product: Array<PlaidProduct>;
+    countryCodes?: Array<string>;
+    language?: string;
+    userEmailAddress?: string;
+    userLegalName?: string;
+    webhook?: string;
+    linkCustomizationName?: string;
+    accountSubtypes?: Array<LinkAccountSubtype>;
+    oauthNonce?: string;
```

Comment:
should oauth fields be wrapped in a `OAuthConfiguration` object like iOS?

Reply:
makes sense, they're iOS only